core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -718,6 +725,33 @@ private void maybeRefreshEngine() {
         }
     }
 
+    private void maybeUpdateGlobalCheckpoints() {
+        if (indexSettings.getGlobalCheckpointInterval().millis() > 0) {
```

Comment:
yeah, this is copy paste from the refresh logic which has to be defensive for dynamic setting update. Not needed here indeed (yet :)) . I removed it.